source/common/event/BUILD

```diff
@@ -40,6 +52,36 @@ envoy_cc_library(
     }),
 )
 
+envoy_cc_library(
+    name = "signal_lib",
+    deps = envoy_cc_platform_dep("signal_impl_lib"),
+)
+
+envoy_cc_posix_library(
+    name = "signal_impl_lib",
+    srcs = ["posix/signal_impl.cc"],
+    hdrs = ["posix/signal_impl.h"],
+    strip_include_prefix = "posix",
```

Comment:
the strip_include_prefix may be a source of confusion, but there is prior art in source/common/common/BUILD and source/common/filesystem/BUILD so I guess this is fine.

I would have suggested having a signal_impl.h that includes the right platform header.